cf-execd/cf-execd.c

```diff
@@ -431,12 +455,29 @@ static void CFExecdMainLoop(EvalContext *ctx, Policy **policy, GenericAgentConfi
             }
         }
         /* 1 Minute resolution is enough */
-        MaybeSleepLog(LOG_LEVEL_VERBOSE, "Sleeping for pulse time %u seconds...", CFPULSETIME);
+        terminate = HandleRequestsOrSleep(CFPULSETIME, "pulse time");
+        if (terminate)
+        {
+            break;
+        }
     }
 }
 
 #else  /* ! __MINGW32__ */
 
+// msg should include exactly one reference to unsigned int.
+static inline unsigned int MaybeSleepLog(LogLevel level, const char *msg, unsigned int seconds)
```

Comment:
I didn't get this at first read. I'd prefer `msg_format` as the parameter name and:

```
// msg_format should include exactly one '%u'
```

Reply:
It's not a function I added, though. I'll add a commit improving it.